### PR TITLE
N°5066 - Clean CMDBSource methods

### PR DESCRIPTION
**Pull request by Hipska**
This improves the function calls to use more relevant `QueryToScalar` instead of the `QueryToCol`.

This also deprecates `GetServerInfo` since the info is not always reliable. (Related to https://github.com/Combodo/itop-system-information/pull/5) (See https://www.php.net/manual/en/mysqli.get-server-info.php#118822)

### CODE REVIEW

File: core/cmdbsource.class.inc.php [hunk 1]
```diff
@@ -350,14 +350,20 @@ public static function IsDB($sSource)
...
 	{
-		$aVersions = self::QueryToCol('SELECT Version() as version', 'version');
-		return $aVersions[0];
+		return static::QueryToScalar('SELECT VERSION()', 0);
 	}
 
 	/**
-	 * @return string
+	 * @deprecated Use `CMDBSource::GetDBVersion` instead.
+	 * @see mysqli_get_server_info
```
Comment (piRGoif) [inline core/cmdbsource.class.inc.php]:
```suggestion
	 * @uses mysqli_get_server_info
```

We prefer to write `@uses` in such case (see https://docs.phpdoc.org/3.0/guide/references/phpdoc/tags/uses.html#uses-used-by)

**Review by piRGoif — CHANGES_REQUESTED** [review]
Thanks for those improvements, seems good to me ! 
Just a little remark about the phpdoc

File: core/cmdbsource.class.inc.php (hunk 1 above)
Comment (Hipska) [reply core/cmdbsource.class.inc.php]: Thanks, didn't know that one.

**Review by piRGoif — DISMISSED** [review]

**Review by piRGoif — APPROVED** [review]

**Review by piRGoif — CHANGES_REQUESTED** [review]
After a team review, we found that the behavior has changed : before, if the result was 0 we were returning an empty string... Can you modify that ? 

I've checked exceptions, seems to me that we are getting the same with \CMDBSource::QueryToArray and \CMDBSource::QueryToScalar

**Comment by Hipska** [comment]
Fixed the conflicts

**Comment by piRGoif** [comment]
Hello,

> After a team review, we found that the behavior as changed : before, if the result was 0 we were returning an empty string... Can you modify that ?

@Hipska you asked for a new review, but this wasn't changed right ?

**Comment by Hipska** [comment]
Ah, you mean for `GetServerVariable`? You want me to return this instead?

```php
return static::QueryToScalar($sSql, 0) ?: '';
```

**Comment by piRGoif** [comment]
> Ah, you mean for `GetServerVariable`? You want me to return this instead?
> 
> ```
> return static::QueryToScalar($sSql, 0) ?: '';
> ```

Would be perfect, thanks !

**Comment by Hipska** [comment]
Applied, please review.

**Review by piRGoif — APPROVED** [review]

**Comment by piRGoif** [comment]
Sorry, we missed technical review today (too much PR to review). Will be reviewed next month !

**Review by Molkobain — APPROVED** [review]

**Comment by Molkobain** [comment]
Accepted during technical review.

**Comment by Molkobain** [comment]
Accepted in functional review, will be part of 3.1.0

**Pull request merged**